src/main/java/io/cryostat/net/reports/ActiveRecordingReportCache.java

```diff
@@ -93,20 +88,15 @@
     Future<String> get(ConnectionDescriptor connectionDescriptor, String recordingName) {
         CompletableFuture<String> f = new CompletableFuture<>();
         try {
-            f.complete(
-                    cache.get(
-                            new SubprocessReportGenerator.RecordingDescriptor(
-                                    connectionDescriptor, recordingName)));
+            f.complete(cache.get(new RecordingDescriptor(connectionDescriptor, recordingName)));
```

Comment:
From this method, I'm assuming reports are saved as String representations? Or is this a String representation of the Path to the file? Also, if ActiveRecordingReportCache#get returns null then a call to ActiveRecordingReportCache#getReport would have to be made to generate the report, correct?

Reply:
> From this method, I'm assuming reports are saved as String representations? Or is this a String representation of the Path to the file?

It is the report HTML itself as a String. The cache is configured with `softValues`, so if the available heap memory in the JVM is low then the garbage collector may reclaim space by removing cached reports. We could write them to disk and cache the paths and then use a cache eviction listener to delete the files on disk, too. That would make the `ActiveReportCache` more similar to the `ArchivedReportCache`, which does write to disk and has no time-based expiration policy (since archived recordings will not change and therefore the reports are always current for a given archived recording once generated), and which returns `Path`s instead of `String`s.

> Also, if ActiveRecordingReportCache#get returns null then a call to ActiveRecordingReportCache#getReport would have to be made to generate the report, correct?

`ActiveRecordingReportCache#get` should never return null - it calls `cache.get()` internally, and `cache.get()` will either return a cached value or will compute the value (by calling `getReport` as configured by line 85), cache it, and return that.